src/controllers/expense.controller.js

```diff
@@ -0,0 +1,136 @@
+'use strict';
+
+const { getAllUsers } = require('../services/user.service');
+const {
+  getAllExpenses,
+  getExpenseById,
+  updateExpense,
+  deleteExpenseById,
+  setAllExpenses,
+  createExpense,
+} = require('../services/expense.service');
+
+function get(req, res) {
+  const allExpenses = getAllExpenses();
+  const { userId, categories, from, to } = req.query;
+  let filteredExpenses = null;
+
+  if (categories && userId) {
+    filteredExpenses = allExpenses.filter(item => {
+      return item.category === categories;
+    });
+
+    return res.send(filteredExpenses);
+  };
+
+  if (userId) {
+    filteredExpenses = allExpenses.filter(item => {
+      return item.userId === +userId;
+    });
+
+    return res.send(filteredExpenses);
+  }
+
+  if (allExpenses.length === 0) {
+    return res.send([]);
+  }
+
+  if (from && to) {
+    filteredExpenses = allExpenses.filter(item => {
+      return item.spentAt > from && item.spentAt < to;
+    });
+
+    return res.send(filteredExpenses);
+  }
+
+  res.send(allExpenses);
+}
+
+function getOneExpense(req, res) {
+  const { id } = req.params;
+  const expense = getExpenseById(+id);
+
+  if (!expense) {
+    res.sendStatus(404);
+
+    return;
+  }
+
+  res.send(expense);
+}
+
+function addExpense(req, res) {
+  const {
+    userId,
+    spentAt,
+    title,
+    amount,
+    category,
+    note,
+  } = req.body;
+
+  if (!title || !category || !note || !amount || !userId || !spentAt) {
+    res.status(400).send('Invalid request body');
+
+    return;
+  }
+
+  const user = getAllUsers().find(u => u.id === +userId);
```

Comment:
Fix naming. u - blurry meaning

Reply:
it's naming inside callback function and I don't use it outside.  guess that it's clear to understand (user => u)